examples/distributed_evaluation.py

```diff
@@ -0,0 +1,191 @@
+
+        def forward(self, x: Tensor) -> Tensor:
+            return self.model(x)
+
+        def test_step(self, batch: Tuple[Tensor, Tensor], batch_idx: int) -> None:
+            preds = model(batch[0])
+            target = batch[1]
+            self.metric.update(preds, target)
+
+        def on_test_epoch_end(self) -> None:
+            self.log("test_acc", self.metric.compute())
+
+        def test_dataloader(self) -> DataLoader:
+            return DataLoader(
+                dataset,
+                batch_size=batch_size,
+                sampler=sampler_class(dataset),
+            )
+
+    model = DummyLightningModule(model)
+
+    trainer = Trainer(
+        devices=num_processes,
+        accelerator="cpu" if not gpu else "gpu",
+    )
+
+    res = trainer.test(model)
+    manual_res = calculate_accuracy_manually(dataset, model)
+    print(manual_res)
+    if torch.allclose(torch.tensor(res[0]["test_acc"]), manual_res):
+        print("success! result matched manual calculation")
+    else:
+        print("failure! result did not match manual calculation")
+
+
+def _use_torch_worker_fn(
+    rank: int, model: Module, dataset: Dataset, batch_size: int, use_standard: bool, num_processes: int, gpu: bool
+) -> None:
+    """Worker function for torch.distributed evaluation."""
+    os.environ["MASTER_ADDR"] = "localhost"
+    os.environ["MASTER_PORT"] = "29500"
+    dist.init_process_group("nvcc" if gpu else "gloo", rank=rank, world_size=num_processes)
+
+    device = torch.device(f"cuda:rank{rank}") if gpu else torch.device("cpu")
+
+    sampler_class = DistributedSampler if use_standard else EvaluationDistributedSampler
+
+    dataloader = DataLoader(
+        dataset,
+        batch_size=batch_size,
+        sampler=sampler_class(dataset, num_processes, rank),
+    )
+
+    metric = torchmetrics.classification.MulticlassAccuracy(num_classes=10, average="micro")
+    metric = metric.to(device)
+
+    batches, num_samples = 0, 0
+    for _, batch in enumerate(dataloader):
+        if gpu:
+            batch = batch.cuda()
+        preds = model(batch[0])
+        target = batch[1]
+
+        metric.update(preds.to(device), target.to(device))
+        num_samples += len(target)
+        batches += 1
+
+    res = metric.compute()
+
+    print(f"Rank {rank} processed {num_samples} samples and {batches} batches and calculated accuracy: {res}")
+
+    manual_res = calculate_accuracy_manually(dataset, model)
+    if torch.allclose(res, manual_res):
+        print("success! result matched manual calculation")
+    else:
+        print("failure! result did not match manual calculation")
+
+
+def use_torch(
+    model: Module, dataset: Dataset, batch_size: int, use_standard: bool, num_processes: int, gpu: bool
+) -> None:
+    """Use torch.distributed to evaluate a model on a dataset."""
+    mp.spawn(_use_torch_worker_fn, nprocs=2, args=(model, dataset, batch_size, use_standard, num_processes, gpu))
+
+
+def main() -> None:
```

Comment:
let's consider using `fire` or `jsonargapsrse` as CLI for examples... :)

Reply:
but then user would need to install those just to run the example